api/controller/pin.js

```diff
@@ -1,88 +1,146 @@
-import { createPin } from "../model";
+import { validationResult } from "express-validator";
+import { createPin, deletePin, getPinById, updatePin } from "../model";
 import User from "../model/user";
 
 const addNewPin = async (req, res) => {
-  const { body = {} } = req;
+  const errors = validationResult(req);
+  if (!errors.isEmpty()) {
+    console.error("Validation errors:", errors.array());
+    return res.status(400).json({ errors: errors.array() });
+  }
```

Comment:
This you can move to middleware only. No need to write in each controller. Our motive is to handle these there only.

Reply:
Done